web-admin/src/features/organizations/settings/FaviconSettings.svelte

```diff
@@ -0,0 +1,154 @@
+<script lang="ts">
+  import { invalidateAll } from "$app/navigation";
+  import {
+    createAdminServiceCreateAsset,
+    createAdminServiceUpdateOrganization,
+    getAdminServiceGetOrganizationQueryKey,
+  } from "@rilldata/web-admin/client";
+  import { CANONICAL_ADMIN_URL } from "@rilldata/web-admin/client/http-client";
+  import SettingsContainer from "@rilldata/web-admin/features/organizations/settings/SettingsContainer.svelte";
+  import { Button } from "@rilldata/web-common/components/button";
+  import Rill from "@rilldata/web-common/components/icons/Rill.svelte";
+  import EditIcon from "@rilldata/web-common/components/icons/EditIcon.svelte";
+  import {
+    Popover,
+    PopoverContent,
+    PopoverTrigger,
+  } from "@rilldata/web-common/components/popover";
+  import { extractFileExtension } from "@rilldata/web-common/features/entity-management/file-path-utils";
+  import { queryClient } from "@rilldata/web-common/lib/svelte-query/globalQueryClient";
+  import { builderActions, getAttrs } from "bits-ui";
+  import FileInput from "@rilldata/web-common/components/forms/FileInput.svelte";
+
+  export let organization: string;
+  export let organizationFaviconUrl: string | undefined;
+
+  $: faviconUrl = organizationFaviconUrl;
+
+  const assetCreator = createAdminServiceCreateAsset();
+  const orgUpdater = createAdminServiceUpdateOrganization();
+
+  let open = false;
+  let assetId = "";
+
+  async function uploadFile(file: File) {
+    const ext = extractFileExtension(file.name);
+    const assetResp = await $assetCreator.mutateAsync({
+      organizationName: organization,
+      data: {
+        type: "image",
+        name: "favicon",
+        extension: ext,
+        cacheable: true,
+        estimatedSizeBytes: file.size.toString(),
+      },
+    });
+
+    await fetch(assetResp.signedUrl, {
+      method: "PUT",
+      body: file,
+      headers: assetResp.signingHeaders,
+    });
+    assetId = assetResp.assetId;
+    return `${CANONICAL_ADMIN_URL}/v1/assets/${assetId}/download`;
+  }
+
+  function onCancel() {
+    open = false;
+    faviconUrl = organizationFaviconUrl;
+  }
+
+  async function removeLogo() {
+    onCancel();
+    await $orgUpdater.mutateAsync({
+      name: organization,
+      data: {
+        faviconAssetId: "",
+      },
+    });
+    void queryClient.invalidateQueries(
+      getAdminServiceGetOrganizationQueryKey(organization),
+    );
+    void invalidateAll();
+  }
+
+  async function onSave() {
+    onCancel();
```

Comment:
I'd expect the `onCancel()` (or a simple `closePopover()`) call to happen _after_ the mutation, not before. Then we we can handle loading & error states in the popover itself.

(And the same goes for the `onCancel()` call in the `removeLogo()` function.)